### ISSUE TITLE
Null reference exception in NativeClient.LoadAchievements funtion

### ISSUE THREAD

**Issue by Beerayco** [playgameservices/play-games-plugin-for-unity#2580]
Hello, we are noticing that after upgrading the plugin to latest version (v0.9.64) we still are experiencing some issues with the new achievements system.

This is the stack trace we receive in the Unity crash report service from some players:

`NullReferenceException: Object reference not set to an instance of an object.
GooglePlayGames.PlayGamesPlatform+<>c__DisplayClass58_0.<LoadAchievements>b__0 (GooglePlayGames.BasicApi.Achievement[] ach)
GooglePlayGames.OurUtils.PlayGamesHelperObject.`

Could you give us some feedback about possible solutions for this?
Thank you for your time guys!

**Comment by olehkuznetsov** [comment]
Are you using ReportProgress function?

**Comment by Beerayco** [comment]
No, we use the UnlockAchievement function whenever we want to give an achievement to a player

**Comment by SeriouslySam** [comment]
We seem to be having the same issue. Also at v0.9.64 on Unity 2017.4.27
It doesn't happen to everyone but seems to happen across different OS versions and different device models. Seeing at least on Android 5, 7, 8 and 9.

Non-fatal Exception: java.lang.Exception: NullReferenceException :  Object reference not set to an instance of an object
       at GooglePlayGames.PlayGamesPlatform+<LoadAchievements>c__AnonStorey5.<>m__0(GooglePlayGames.PlayGamesPlatform+<LoadAchievements>c__AnonStorey5)
       at GooglePlayGames.Native.NativeClient+<InvokeCallbackOnGameThread>c__AnonStorey3`1[GooglePlayGames.BasicApi.Achievement[]].<>m__0(GooglePlayGames.Native.NativeClient+<InvokeCallbackOnGameThread>c__AnonStorey3`1[GooglePlayGames.BasicApi.Achievement[]])
       at GooglePlayGames.OurUtils.PlayGamesHelperObject.Update(GooglePlayGames.OurUtils.PlayGamesHelperObject)

**Comment by sp-ivan-hernandez** [comment]
We are also having this issue at v0.9.64, and Unity 2018.4.3.
But it is not always happening, it depends on the user that triggers the LoadAchievements.

Any plans on fixing it?

**Comment by tommikiviniemi-srs** [comment]
Hi,
This is one of the main crashes in our game, so what is the status of this bug -- please advice on how to proceed? Is there a version we can update to to resolve the situation, or is there some other workaround?

Thanks.

**Comment by olehkuznetsov** [comment]
If you are using ReportProgress method try to replace it by UnlockAchievement or IncrementAchievement

**Comment by vaibhavgpt-google** [comment]
Closing this issue as part of a cleanup of older tickets. Google Play Games v1 is now deprecated. Please upgrade to version v2.1.0 or newer: https://github.com/playgameservices/play-games-plugin-for-unity/releases/tag/v2.1.0. If this issue persists on the new v2 architecture, please open a new ticket.
